File: client/java/src/main/java/com/be/mesh/client/mpc/Execution.java
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */
package com.be.mesh.client.mpc;

/**
 * @author coyzeng@gmail.com
 */
public interface Execution<T> extends Invoker<T> {

    /**
     * Execution schema.
     */
    T schema();

    /**
     * Inspect execution.
     */
    Inspector inspect();

    /**
     * Execution input type.
     */
    <I extends Parameters> Types<I> intype();

    /**
     * Execution output return type.
     */
    <O extends Returns> Types<O> retype();

    /**
     * Reflect input type.
     */
    Parameters inflect();

    /**
     * Reflect output return type.
     */
    Returns reflect();
}

File: client/golang/mpc/execution.go
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */

package mpc

import (
	"github.com/be-io/mesh/client/golang/macro"
)

type Generic interface {
	GetURN() string
	GetProto() string
	GetCodec() string
	GetTimeout() int64
	GetRetries() int
}

type Execution interface {

	// Schema is the execution schema.
	Schema() Generic

	// Inspect execution.
	Inspect() macro.Inspector

	Invoker
}
